uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,447 @@ define([
+    //leaveRequests.filter - to handle left nav filter data
+    vm.leaveRequests = {
+      table: {
+        list: []
+      },
+      filter: {
+        list: []
+      }
+    };
+    vm.filters = {
+      contactFilters: {
+        region: null,
+        department: null,
+        level_type: [],
+        location: null
+      },
+      leaveRequestFilters: {
+        selectedUsers: null,
+        selectedPeriod: null,
+        selectedAbsenceTypes: null,
+        leaveStatus: allStatus,
+        pending_requests: false
+      }
+    };
+    vm.pagination = {
+      page: 1,
+      size: 7
+    };
+    vm.loading = {
+      content: true,
+      page: true
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+    vm.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Returns the label of a Leave request when id is given
+     *
+     * @param {string} id - id of the leave request
+     * @return {string}
+     */
+    vm.getLeaveStatusByValue = function (id) {
+      if (vm.leaveRequestStatuses.length) {
+        return vm.leaveRequestStatuses.find(function (status) {
+          return status.value == id;
+        }).label;
+      }
+    };
+
+    /**
+     * Returns the title of a Absence type when id is given
+     *
+     * @param {string} id - id of the Absence type
+     * @return {string}
+     */
+    vm.getAbsenceTypesByID = function (id) {
+      if (vm.absenceTypes && id) {
+        var type,
+          i;
+        for (i in vm.absenceTypes) {
+          type = vm.absenceTypes[i];
+          if (type.id == id) {
+            return type.title;
+          }
+        }
+      }
+    };
+
+    /**
+     * Filters leave requests by status
+     *
+     * @param {Object} status - status object
+     * @return {array}
+     */
+    vm.filterLeaveRequestByStatus = function (status) {
+      if (status.name === 'all' || status === '') {
+        return vm.leaveRequests.filter.list;
+      }
+
+      return vm.leaveRequests.filter.list.filter(function (request) {
+        return request.status_id == status.value;
+      })
+    };
+
+    /**
+     * Returns the class name for filter navigation when name is given
+     *
+     * @param {string} name - name of the status
+     * @return {string}
+     */
+    vm.getNavBadge = function (name) {
+      if (name === 'approved') {
+        return 'badge-success';
+      } else if (name === 'rejected') {
+        return 'badge-danger';
+      } else if (name === 'cancelled' || name === 'all') {
+        return '';
+      } else {
+        return 'badge-primary';
+      }
+    };
+
+    /**
+     * Returns an array of a given size
+     *
+     * @param {number} n - no of elements in the array
+     * @return {Array}
+     */
+    vm.range = function (n) {
+      if (n) {
```

Comment:
I think this could be more elegant, just return an Array of length 0 if the param is missing
```js
vm.range = function (n) {
  return new Array(n || 0);
}
```